pkg/commands/git_commands/commit_loader.go

```diff
@@ -28,29 +29,28 @@ type CommitLoader struct {
 	*common.Common
 	cmd oscommands.ICmdObjBuilder
 
-	getCurrentBranchInfo func() (BranchInfo, error)
-	getRebaseMode        func() (enums.RebaseMode, error)
-	readFile             func(filename string) ([]byte, error)
-	walkFiles            func(root string, fn filepath.WalkFunc) error
-	dotGitDir            string
+	getRebaseMode   func() (enums.RebaseMode, error)
+	readFile        func(filename string) ([]byte, error)
+	walkFiles       func(root string, fn filepath.WalkFunc) error
+	dotGitDir       string
+	doBranchesExist map[string]bool
 }
 
 // making our dependencies explicit for the sake of easier testing
 func NewCommitLoader(
 	cmn *common.Common,
 	cmd oscommands.ICmdObjBuilder,
 	dotGitDir string,
-	getCurrentBranchInfo func() (BranchInfo, error),
 	getRebaseMode func() (enums.RebaseMode, error),
 ) *CommitLoader {
 	return &CommitLoader{
-		Common:               cmn,
-		cmd:                  cmd,
-		getCurrentBranchInfo: getCurrentBranchInfo,
-		getRebaseMode:        getRebaseMode,
-		readFile:             os.ReadFile,
-		walkFiles:            filepath.Walk,
-		dotGitDir:            dotGitDir,
+		Common:          cmn,
+		cmd:             cmd,
+		getRebaseMode:   getRebaseMode,
+		readFile:        os.ReadFile,
+		walkFiles:       filepath.Walk,
+		dotGitDir:       dotGitDir,
+		doBranchesExist: make(map[string]bool),
```

Comment:
Rather than store a map of all the candidate branches and whether they exist, can we instead store the list of branches which do exist? So we can check if the slice is nil whenever we go to load commits and if it is, we can work out what the existing branches are. Having a per-branch mapping (with per-branch caching) seems too granular to me.

Reply:
Good point. I went one step further and now cache the concatenated string of quoted branches instead of doing the quoting every time.